test/EFCore.MySql.FunctionalTests/Startup.cs

```diff
@@ -47,15 +47,40 @@ public static void ConfigureEntityFramework(IServiceCollection services, DbConne
         {
             if (connection == null)
             {
-                services.AddDbContextPool<AppDb>(
-                    options => options.UseMySql(AppConfig.Config["Data:ConnectionString"],
-                        mysqlOptions => mysqlOptions.MaxBatchSize(AppConfig.EfBatchSize)));
+                if (AppConfig.EfRetryOnFailure > 0)
+                {
+                    services.AddDbContextPool<AppDb>(
+                        options => options.UseMySql(AppConfig.Config["Data:ConnectionString"],
+                            mysqlOptions => 
+                            {
+                                mysqlOptions.MaxBatchSize(AppConfig.EfBatchSize);
+                                mysqlOptions.EnableRetryOnFailure(AppConfig.EfRetryOnFailure, TimeSpan.FromSeconds(5), null);
```

Comment:
Need enable user input a retry timespan range, a fixed value is not good for retry.

Reply:
Normally the user does configure this. It's just configured to 5 seconds for the functional tests which is fine.

Reply:
Correct, the user can specify any timespan.

These options match exactly with sql server and PG

Reply:
No, my means is the time should not be fixed, user could provide 5~8seconds timespan, and ef will retry a random seconds in the 5~8 range.

Reply:
I did some research on how the base EF Core libraries work with a registered ExecutionStrategy, the timeout setting actually does what you are outlining and then some. The maxRetryDelay is not actually the time between retries, it is the total time to retry the operation. The ExecutionStrategy class using this max amount, adds a little randomness and then spreads out the retries so there are more at the start and then exponentially it increases the time between retries. So for the most part we are good to go, except for the initial connection (see below).

What I have to work on is the custom routine I had to do (src/EFCore.MySql/Internal/MySqlRetryNoDependendiciesExecutionStrategy.cs)  to handle the connection that was outside of the ExecutionStrategy implementation in the src/EFCore.MySql/Storage/Internal/MySqlConnectionSettings.cs class. I used the maxRetryDelay as a time between retries as I thought that is what it was. I will correct.

I will post another pull request for this, I'm going to work on it now.

See the base EF Core ExecutionStrategy.cs for reference.
https://github.com/aspnet/EntityFrameworkCore/blob/09c9b263d648dac4432b9dfb280f84f651c33949/src/EFCore/Storage/ExecutionStrategy.cs

Reply:
just submitted pull request for exponential backoff.

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/pull/431